examples/io_test/io_bench.f90

```diff
@@ -55,6 +66,7 @@ program io_bench
    end if
 
    deallocate (u1)
+   call decomp_2d_io_fin
```

Comment:
Could we call this from `decomp_2d_finalize`?

Reply:
As a side remark, this might introduce circular dependencies. Currently, decomp_2d does not depends on decomp_2d_io.

Reply:
I'd be happy to move init and finalize as suggested, however if it introduces circular dependencies obviously this is not an option.

Also I remember some discussion about being able to use decomp2d without MPI and now you've pointed it out I think even without circular dependencies this would prevent that.

In which case I suggest we leave as is.